src/parse/asp/builtins.go

```diff
@@ -233,12 +233,18 @@ func getRuleMetadata(s *scope, args []pyObject) pyObject {
 	label := core.ParseBuildLabelContext(name, s.pkg)
 
 	t := s.state.Graph.Target(label)
+
 	if t == nil {
 		if label.Subrepo == s.pkg.SubrepoName && label.PackageName == s.pkg.Name {
 			// This is a get_rule_metadata in the same package, check the target exists.
 			log.Fatalf("Target %s is not defined in this package yet; it has to be defined before the get_rule_metadata() call", name)
 		}
-		t = s.state.WaitForBuiltTarget(label, core.NewBuildLabel(s.pkg.Name, "all"))
+		func(){
```

Comment:
I think these two are common enough that you could pull them out to a separate function even

Reply:
Done